core_lib/src/interface/editor.cpp

```diff
@@ -863,6 +864,208 @@ bool Editor::importImage(QString filePath)
     }
 }
 
+bool Editor::importMovieVideo(QString filePath, int fps, QProgressDialog &progress)
+{
+    Layer* layer = layers()->currentLayer();
+    if (layer->type() != Layer::BITMAP)
+    {
+        return false;
+    }
+
+    qDebug() << "-------IMPORT MOVIE VIDEO------" << filePath;
+
+    // --------- Import all the temporary frames ----------
+    QTemporaryDir tempDir;
+    if (!tempDir.isValid())
+    {
+        qDebug() << "Could not create a temporary folder";
+        return false;
+    }
+
+    QString ffmpegPath = ffmpegLocation();
+    if (!QFile::exists(ffmpegPath))
+    {
+        qDebug() << "Please place ffmpeg.exe in plugins directory";
+        return false;
+    }
+
+    // Get frame estimate
```

Comment:
You make a frame estimate here. I think it should be used to limit the amount of frames to be imported.
I imported a mp4, that lasted 13 seconds. It took a minute.
After that I tried to import a mp4 that lasted 3:56 minutes. After 4-5 minutes it started importing pictures. After importing about 350 frames, the software crashed.
What is the biggest video you've tried to import?
Shouldn't we set a maximum of 300-500 frames?

Reply:
There is no technical limit. If you are on Windows, it probably crashed because you ran out of memory, but that's entirely dependent on the system and is difficult for us to set a good limit on. I can't test this right now, but I will later.

One solution is to output the frame from ffmpeg directly to the temporary folder (or moving files into the temporary folder), which prevents having to have all the frames loaded into memory and would cut the import time in half, however this would not be good for pcl files where writing to the temporary folder is practically the equivalent of saving.

Reply:
We could load in chunks at a time, since we can't rely on people understanding that their video might generates thousands of images, this will still take a hit in loading time but imo. it's better than crashing the program and won't be as bad as moving frames to the temporary folder for each frame.